magento/devdocs
# Add rake task for image optimization

## Purpose of this pull request



This pull request add a `rake image_opt` task that runs the [image_opt](https://github.com/toy/image_optim) tool for a list of modified files, or for the particular path if provided with `path` argument.

## Testing

Update your gem bundle with `bundle install` to add the tool

1. Add one or more images to the project
2. Run `rake image_optim`

Expected result: the new image/images are processed and compressed.

Example:

```
~/Projects/repos/magento/devdocs (ds_image-compr-plugin *)$ rake image_optim
bin/image_optim is not an image or there is no optimizer for it
Gemfile is not an image or there is no optimizer for it
Gemfile.lock is not an image or there is no optimizer for it
Rakefile is not an image or there is no optimizer for it
optimizing: 100.0% (elapsed: 24s)
48.60% 148.5K  Screen-Shot-2019-05-15-at-14.41.09.png
------         common/images/b2b/payment-on-credit.png
------         common/images/cloud/cloud-fastly-customize-503-response.png
Total: 48.58% 148.5K
```

3. Run `rake image_optim path=mftf` to test image compression at a particular directory
